app/src/main/java/ch/epfl/polychef/users/SearchUser.java

```diff
@@ -0,0 +1,89 @@
+package ch.epfl.polychef.users;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import ch.epfl.polychef.CallHandler;
+import ch.epfl.polychef.Miniatures;
+import ch.epfl.polychef.recipe.Recipe;
+import ch.epfl.polychef.recipe.RecipeStorage;
+import ch.epfl.polychef.utils.Similarity;
+
+public class SearchUser {
+
+    private static final String TAG = "SearchUser";
+
+    private static final SearchUser INSTANCE = new SearchUser();
+
+    private SearchUser() {
+    }
+
+    public static SearchUser getInstance() {
+        return INSTANCE;
+    }
+
+    public void searchForUser(String query, CallHandler<List<Miniatures>> caller) {
+        searchUser(query, this::compareSimilarity, caller);
+    }
+
+    private void searchUser(String query, BiFunction<String, User, Boolean> comparator, CallHandler<List<Miniatures>> caller) {
+        DatabaseReference nameRef = getDatabase().getReference(UserStorage.DB_NAME);
+        nameRef.addValueEventListener(new ValueEventListener() {
```

Comment:
Do we really need `addValueEventListener` here? Wouldn't a `addListenerForSingleValueEvent` be enough?